satpy/readers/netcdf_utils.py

```diff
@@ -117,14 +128,22 @@ def __init__(self, filename, filename_info, filetype_info,
             self.collect_dimensions("", file_handle)
         self.collect_cache_vars(cache_var_size)
 
-        if cache_handle:
-            self.file_handle = file_handle
-        else:
+        if not cache_handle:
             file_handle.close()
 
     def _get_file_handle(self):
         return netCDF4.Dataset(self.filename, "r")
 
+    @property
+    def file_handle(self):
+        """Backward-compatible way for file handle caching."""
+        warnings.warn(
+                "attribute .file_handle is deprecated, use .manager instead",
+                DeprecationWarning)
+        if self.manager is None:
+            return None
+        return self.manager.acquire()
+
     @staticmethod
     def _set_file_handle_auto_maskandscale(file_handle, auto_maskandscale):
         if hasattr(file_handle, "set_auto_maskandscale"):
```

Comment:
Not that this has to be handled in your PR, but if I remember correctly this `Dataset`-level `set_auto_maskandscale` was added to netcdf4-python quite a while ago. It seems error prone and confusing to silently call the method only if it exists and to not log/inform the user that it wasn't used when it was expected. Maybe we should remove this method on the file handler class and always call `file_handle.set_auto_maskandscale` no matter what. Your wrapper does it already.